lib/lightning/auditing.ex

```diff
@@ -4,17 +4,83 @@ defmodule Lightning.Auditing do
   """
 
   import Ecto.Query
+
   alias Lightning.Accounts.User
   alias Lightning.Auditing.Audit
   alias Lightning.Repo
+  alias Lightning.VersionControl.ProjectRepoConnection
+  alias Lightning.Workflows.Trigger
 
   def list_all(params \\ %{}) do
     from(a in Audit,
+      left_join: p in ProjectRepoConnection,
+      on: p.id == a.actor_id and a.actor_type == :project_repo_connection,
+      left_join: t in Trigger,
+      on: t.id == a.actor_id and a.actor_type == :trigger,
       left_join: u in User,
-      on: [id: a.actor_id],
-      select_merge: %{actor: u},
+      on: u.id == a.actor_id and a.actor_type == :user,
+      select_merge: %{
+        project_repo_connection: p,
+        trigger: t,
+        user: u
+      },
       order_by: [desc: a.inserted_at]
     )
     |> Repo.paginate(params)
+    |> then(fn %{entries: entries} = result ->
+      Map.put(
+        result,
+        :entries,
+        Enum.map(entries, &Map.put(&1, :actor_display, actor_display_for(&1)))
```

Comment:
There is a strong argument to be made for having the generation of the `actor_display` live in the view, but my counterargument is that currently this query is only used by the view and that it is much easier to test.